extensions/amp-analytics/0.1/cid-impl.js

```diff
@@ -285,26 +275,33 @@ function getBaseCid(cid, persistenceConsent) {
  * @param {string} cidString Actual cid string to store.
  */
 function store(win, cidString) {
-  try {
-    const item = {
-      time: Date.now(),
-      cid: cidString,
-    };
-    const data = JSON.stringify(item);
-    win.localStorage.setItem('amp-cid', data);
-  } catch (ignore) {
-    // Setting localStorage may fail. In practice we don't expect that to
-    // happen a lot (since we don't go anywhere near the quota, but
-    // in particular in Safari private browsing mode it always fails.
-    // In that case we just don't store anything, which is just fine.
+  const item = {
+    time: Date.now(),
+    cid: cidString,
+  };
+  const data = JSON.stringify(item);
+  const viewer = viewerFor(win);
+  if (viewer.isIframed()) {
+    // If we are being embedded, try to save the base cid to the viewer.
+    viewer.baseCid(data);
```

Comment:
I hate overloading this API. Why not keep `getBaseCid` and add a `storeBaseCid`?